pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -133,7 +133,17 @@ public CompletableFuture<SchemaAndMetadata> getSchema(String schemaId, SchemaVer
                     } else {
                         return Functions.bytesToSchemaInfo(stored.data)
                                 .thenApply(Functions::schemaInfoToSchema)
-                                .thenApply(schema -> new SchemaAndMetadata(schemaId, schema, stored.version));
+                                .thenApply(schema -> new SchemaAndMetadata(schemaId, schema, stored.version))
+                                .thenApply((schema) -> {
+                                    if (version == SchemaVersion.Latest) {
```

Comment:
Please add a test to show the changes make sense

Reply:
changes are covered in `SchemaServiceTest.java`

Reply:
> changes are covered in `SchemaServiceTest.java`

@nikam14 Please add a test that shows that this change was needed. I don't see any test changes in this PR, so that is missing.